main.c

```diff
@@ -414,41 +412,89 @@ void animationCallback(GPTDriver* _driver) {
   (void)_driver;
 
   // If the foreground is set we skip the animation as a way to avoid it overrides the foreground
-  if(!is_foregroundColor_set) {
+  if(!is_foregroundColor_set && profiles[currentProfile].animationSpeed[currentSpeed] > 0) {
     profiles[currentProfile].callback(ledColors, ledIntensity);
   }
 }
 
 inline void sPWM(uint8_t cycle, uint8_t currentCount, uint8_t start, ioline_t port) {
-  if (start+cycle>0xFF) start = 0xFF - cycle;
+  if (start > 0xFF - cycle || cycle > 0xFF - start) {
```

Comment:
I believe there is a possible overflow when checking `start + cycle > 0xFF` since everything is `uint8_t`. Using a safer alternative.

Reply:
OK